### PR TITLE
[SHACK-377] User config file listener

### PR DESCRIPTION
**Pull request by jonsmorrow**
This change adds a file listener so when a user makes changes
to their config file the app can know about them and update/cancel
update checks/channel.

Signed-off-by: Jon Morrow <jmorrow@chef.io>

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -158,6 +160,18 @@ app.on('do-update-check', (_requestFromUser) => {
   mixlibInstallUpdater.checkForUpdates(workstation.getVersion());
 });
 
+app.on('user-config-update', () => {
+  if (workstation.areUpdatesEnabled()) {
```
Comment (marcparadise) [inline main.js]: Other things can change here, can we include the changed attribute in the event and drive off of that instead?

Comment (jonsmorrow) [reply main.js]: good call.

**Comment by jonsmorrow** [comment]
This change was a bit of an experiment to see how it might work. out of scope fro ga.

**Comment by tyler-ball** [comment]
We're going to consider this a spike but the current code is out of date now. Closing this but it is linked from the feature flag story since it overlaps with desired functionality there. When we implement that we can look at the spike here and incorporate it into the current code base.

**Pull request closed without merging**